src/Tribe/Widget/Widget_Interface.php

```diff
@@ -0,0 +1,226 @@
+
+	/**
+	 * Updates a particular instance of a widget.
+	 *
+	 * This function should check that `$new_instance` is set correctly. The newly-calculated
+	 * value of `$instance` should be returned. If false is returned, the instance won't be
+	 * saved/updated.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $new_instance New settings for this instance as input by the user via
+	 *                                          WP_Widget::form().
+	 * @param array<string,mixed> $old_instance Old settings for this instance.
+	 *
+	 * @return array<string,mixed> Settings to save or bool false to cancel saving.
+	 */
+	public function update( $new_instance, $old_instance );
+
+	/**
+	 * Outputs the settings update form.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $instance Current settings.
+	 *
+	 * @return string Default return is 'noform'.
+	 */
+	public function form( $instance );
+
+	/**
+	 * Returns the widget slug that allows the widget to be built via the widget class by slug.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The widget slug.
+	 */
+	public function get_registration_slug();
+
+	/**
+	 * Sets the aliased arguments array.
+	 *
+	 * @see Tribe__Utils__Array::parse_associative_array_alias() The expected format.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $alias_map An associative array of aliases: key as alias, value as mapped canonical.
+	 *                         Example: [ 'alias' => 'canonical', 'from' => 'to', 'that' => 'becomes_this' ]
+	 */
+	public function set_aliased_arguments( array $alias_map );
+
+	/**
+	 * Gets the aliased arguments array.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,string> The associative array map of aliases and their canonical arguments.
+	 */
+	public function get_aliased_arguments();
+
+	/**
+	 * Returns the arguments for the widget parsed correctly with defaults applied.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $arguments Set of arguments passed to the Widget at hand.
+	 *
+	 * @return array<string,mixed> The parsed widget arguments map.
+	 */
+	public function parse_arguments( array $arguments );
+
+	/**
+	 * Returns the array of arguments for this widget after applying the validation callbacks.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $arguments Set of arguments passed to the Widget at hand.
+	 *
+	 * @return array<string,mixed> The validated widget arguments map.
+	 */
+	public function validate_arguments( array $arguments );
+
+	/**
+	 * Returns the array of callbacks for this widget's arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> A map of the widget arguments that have survived validation.
+	 */
+	public function get_validated_arguments_map();
+
+	/**
+	 * Returns a widget arguments after been parsed.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> The widget arguments, as set by the user in the widget string.
+	 */
+	public function get_arguments();
+
+	/**
+	 * Returns a widget arguments after been parsed.
```

Comment:
```suggestion
	 * Get a widget's arguments after they have been been parsed.
```
Also, by "parsed" - do we mean "filtered"?